### PR TITLE
Add DriveInfo class

### PR DESCRIPTION
**Pull request by josesimoes**
## Description
- Add class code and supporting enums.
- Marked `Directory.GetLogicalDrives()` as obsolete.
- Bump native assembly version.

## Motivation and Context
- Adds support for getting details on a drive, list available drives and format drives.
- Addresses nanoFramework/Home#1229
- `Directory.GetLogicalDrives()` is replaced with `DriveInfo.GetDrives()` to follow .NET API.

## Types of changes

- [ ] Improvement (non-breaking change that improves a feature, code or algorithm)
- [ ] Bug fix (non-breaking change which fixes an issue with code or algorithm)
- [x] New feature (non-breaking change which adds functionality to code)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Config and build (change in the configuration and build system, has no impact on code or features)
- [ ] Dependencies (update dependencies and changes associated, has no impact on code or features)
- [ ] Unit Tests (add new Unit Test(s) or improved existing one(s), has no impact on code or features)
- [ ] Documentation (changes or updates in the documentation, has no impact on code or features)

## Checklist:

- [x] My code follows the code style of this project (only if there are changes in source code).
- [x] My changes require an update to the documentation (there are changes that require the docs website to be updated).
- [ ] I have updated the documentation accordingly (the changes require an update on the docs in this repo).
- [x] I have read the [CONTRIBUTING](https://github.com/nanoframework/.github/blob/main/CONTRIBUTING.md) document.
- [ ] I have tested everything locally and all new and existing tests passed (only if there are changes in source code).
- [ ] I have added new tests to cover my changes.

### CODE REVIEW

**Comment by josesimoes** [comment]
@nfbot updatedependencies

**Comment by josesimoes** [comment]
@nfbot updatedependencies

**Review by Ellerbach — COMMENTED** [review]
Just a question on the DriveType enum, rest seems good and aligned with .NET

File: System.IO.FileSystem/DriveType .cs [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+// See LICENSE file in the project root for full license information.
+//
+
+namespace System.IO
+{
+    /// <summary>
+    /// Defines constants for drive types, including CDRom, Fixed, Network, NoRootDirectory, Ram, Removable, and Unknown.
+    /// </summary>
+    public enum DriveType
+    {
```
[the remark this answers is not in the document]
Comment (josesimoes) [reply System.IO.FileSystem/DriveType .cs:12]: This is aligned. I've just left out the ones that are clearly not available here, like Cdrom and unc share

[the remark this answers is not in the document]
Comment (Ellerbach) [reply System.IO.FileSystem/DriveType .cs:12]: Why are you taking them out? Share can exists at some point, I would rather keep things here, so it's fully aligned

**Review by Ellerbach — APPROVED** [review]
Approving anyway as the enum can be adjusted at any time without impact.

File: System.IO.FileSystem/DriveType .cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (josesimoes) [reply System.IO.FileSystem/DriveType .cs:12]: I can add those, of course. Just trying to save a few bytes in the PE size... 😅
(and I very much doubt we can squeeze a network file share client on our devices 😉)

**Pull request merged**